apps/meteor/server/routes/avatar/utils.ts

```diff
@@ -119,27 +121,39 @@ export async function userCanAccessAvatar({ headers = {}, query = {} }: IIncomin
 	return isAuthenticated;
 }
 
-const getFirstLetter = (name: string) =>
-	name
-		.replace(/[^A-Za-z0-9]/g, '')
-		.substr(0, 1)
-		.toUpperCase();
+const getFirstLetter = (name: string, regExp: RegExp) => {
+	const pattern = regExp || defaultPattern;
+	const sanitizedName = name.replace(/[&<>]/g, '');
+	return sanitizedName.replace(pattern, '').substring(0, 1).toUpperCase();
+};
 
-export const renderSVGLetters = (username: string, viewSize = 200) => {
+export const renderSVGLetters = (roomOrUsername: string, viewSize = 200, req?: IIncomingMessage) => {
 	let color = '';
 	let initials = '';
 
-	if (username === '?') {
+	if (roomOrUsername === '?') {
 		color = '#000';
-		initials = username;
+		initials = roomOrUsername;
 	} else {
-		color = getAvatarColor(username);
-		initials = getFirstLetter(username);
+		const settingsRegExp = (() => {
+			if (!req?.url) return null;
+			return req.url.startsWith('/room')
+				? (settings.get('UTF8_Channel_Names_Validation') as string)
+				: (settings.get('UTF8_User_Names_Validation') as string);
```

Comment:
I don't think this should be necessary. These settings validate username and room name when they are created, so roomOrUsername will be already respecting them. You'd also not need to check the URL in this case.

Was there a speicfic need to do this I might've missed?

Reply:
It was a futile consideration.
Changed to be concise and only perform sanitization process.